### PR TITLE
Bindable plans

### LINKED ISSUE #814: Add bindable field for Plan
Plans can override the bindable attribute of services; our `Plan` type does not currently have this field.

### PR DESCRIPTION
**Pull request by pmorie**
Closes #814; still needs some minor godoc changes but it's late, putting up for preliminary review.

### CODE REVIEW

**Comment by arschles** [comment]
@pmorie looks good so far. I've added the `in-progress` label since you said there's still some godoc changes to do. Let me know when this is ready for final review

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -1369,6 +1369,18 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
 	return nil
 }
 
+// isPlanBindable returns whether the given ServiceClass and ServicePlan
+// combination is bindable.  Plans may override the service-level bindable
+// attribute, so if the plan provides a value, return that value.  Otherwise,
+// return the Bindable field of the ServiceClass.
+func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServicePlan) bool {
```
Comment (MHBauer) [inline pkg/controller/controller.go]: This needs a warning somehow. It's not doing anything complicated, but in theory the serviceplan must 'belong to' the serviceclass. This function does not check that relationship.

My gut says this should be a method on service class that takes a plan, but I'm not sure that's worth it.

I don't think there is any action here.

Comment (pmorie) [inline pkg/controller/controller.go]: I agree, but that is enforced by the code calling this method currently.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -1369,6 +1369,21 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
 	return nil
 }
 
+// isPlanBindable returns whether the given ServiceClass and ServicePlan
+// combination is bindable.  Plans may override the service-level bindable
+// attribute, so if the plan provides a value, return that value.  Otherwise,
+// return the Bindable field of the ServiceClass.
+//
+// Note: enforcing that the plan belongs to the given service class is the
```
Comment (MHBauer) [inline pkg/controller/controller.go]: :ok_hand:

File: pkg/controller/controller_test.go [hunk 3]
```diff
@@ -2559,6 +2661,100 @@ func TestCatalogConversionMultipleServiceClasses(t *testing.T) {
...
+      },
+      {
+        "name": "unbindable-bindable",
+        "id": "s2_plan2_id",
+        "bindable": true
+      }]
+    }
+]}`
+
+func truePtr() *bool {
```
Comment (MHBauer) [inline pkg/controller/controller_test.go]: these ptr functions, eww.

Comment (pmorie) [inline pkg/controller/controller_test.go]: yeah.

**Pull request merged**